python/graphstorm/model/node_gnn.py

```diff
@@ -200,13 +203,22 @@ def append_to_dict(from_dict, to_dict):
             label = data.get_labels(seeds)
             if return_label:
                 append_to_dict(label, labels)
+
+            # pred can be a Tensor or a dict of Tensor
+            # emb can be a Tensor or a dict of Tensor
             if isinstance(pred, dict):
                 append_to_dict(pred, preds)
+            else:
+                assert len(seeds) == 1, \
+                    f"Expect prediction results of multiple node types {label.keys()}" \
+                    f"But only get results of one node type"
+                ntype = list(seeds.keys())[0]
+                append_to_dict({ntype: pred}, preds)
```

Comment:
is this possible that `emb` is a dict while `pred` is a tensor?

Reply:
We just assume, the return value of model.predict can be a tensor or a dict and we turn them into dicts.